x-team/auto-ui
# Add EmailFeed UI component

**Release Type:** *Non-Breaking Feature* 


Fixes https://x-team-internal.atlassian.net/browse/XP-2588

## Description

> A follow-up for the https://x-team-internal.atlassian.net/browse/XP‌-2589 task. Once we have component, representing single conversation item, we should create another UI component to represent a feed of email items similar to existing NotesFeed component.
>
> For the mockup please refer to the epic https://x-team-internal.atlassian.net/browse/XP‌-2625

## Checklist

**Before submitting a pull request,** please make sure the following is done:



## Related PRs



branch | PR
------ | ------
XP-2589-single-email-item | [Auto UI](https://github.com/x-team/auto-ui/pull/332)

## Steps to Test or Reproduce
1- Go to http://localhost:9001/?selectedKind=UI%20Components%2FEmailFeed&selectedStory=initial%20expanded%20all&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel&background=%23fff
2- Confirm that match with the design: https://www.figma.com/file/MEqB5f6kDR0BrPKgxK5dOi/X-Team---XP---Email-History---15th-Jan?node-id=768%3A965
3- Confirm all these features are in the component:

- By default, only load the 3 most recent e-mails. Clicking 'View 3 More Email' shows 3 more emails associated to this e-mail.
- ~~After clicking the above to load 3 more emails, the button is changed to "View 10 More Emails" and remains at 10 to allow loading the remaining emails associated to this applicant in batches of 10~~
- The "Expand All" toggle allows the display of emails in full, whereas the default is toggled off and just shows the shorter form: https://d.pr/free/i/TKW6nj incl subject, from, to and timestamp
- Click "Expand All" shows the full content, incl the above and the body text for emails shown and the ones loaded on clicking the "View X More" button
- "Expand Single Email:" Expand single email when is clicked anywhere within this outlined region here: https://d.pr/free/i/itt1kB



## Screenshots

![email_list_module](https://user-images.githubusercontent.com/5392202/51771000-03bb6a80-20be-11e9-86bb-02d56edfa585.PNG)
